.github/workflows/ci.yml

```
@@ -0,0 +1,98 @@
+name: Django CI
+
+on: [push, pull_request]
+
+env:
+  SECRET_KEY: 'django-insecure-test-key-for-ci-1234567890'
+  POSTGRES_DB: 'test_db'
+  POSTGRES_USER: 'test_user'
+  POSTGRES_PASSWORD: 'test_password'
+  HOST: 'localhost'
+  PORT: '5432'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install flake8
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 
+          
+      - name: Run flake8
+        run: flake8 .
+
+  test:
+    runs-on: ubuntu-latest
+    services:
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pytest pytest-django pytest-cov
+
+      - name: Run tests
+        env:
+          SECRET_KEY: ${{ env.SECRET_KEY }}
+        run: |
+          python manage.py test
+
+
+  build:
+    needs: test
```

Не нужно приступать к сборке, пока тесты и линтеры не прошли
```suggestion
    needs: [test, lint]
```